core/libs/utils/Csv.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * Projects4Me Copyright (c) 2017. Licensing : http://legal.projects4.me/LICENSE.txt. Do not remove this line
+ */
+
+namespace Gaia\Libraries\Utils;
+
+/**
+ * This class provides utility functions for working with CSV files.
+ *
+ * @class Csv
+ * @package Gaia\Libraries\Utils
+ * @author Rana Nouman <ranamnouman@gmail.com>
+ */
+class Csv
+{
+    /**
+     * This function generates CSV files for the provided module data and its relationships. It creates a main CSV file
+     * for the module and additional CSV files for any pending relationships. The main CSV file for the module contains
+     * the relationships of the type hasOne and belongsTo. The additional CSV files contain the relationships of the type
+     * hasMany and manyToMany.
+     *
+     * @param string $moduleName The name of the module.
+     * @param array  $data       The data to be exported.
+     * @param array  $metadata   Metadata information about the module fields.
+     * @param array  $rels       Relationship data for the module.
+     * @param object $model      The model instance.
+     *
+     * @return array An array of file paths to the generated CSV files.
+     */
+    public static function prepareCsvFiles($moduleName, $data, $metadata, $rels, $model)
+    {
+        $modulePluralizedName = strtolower($moduleName) . 's';
+        $csvFiles = [];
+        $modelData = [];
+        $relsData = [];
+        $modelFields = array_keys($metadata['fields']);
+        $csvPath = APP_PATH . DS . 'filesystem' . DS . 'exports' . DS . $modulePluralizedName . '.csv';
```

Comment:
lets store them in a "temp" folder, so that if we need to cleanup later, we can.